### PR TITLE
Wait ready event to set application as started

### PR DESCRIPTION
**Pull request by vmarchaud**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2353
| License       | MIT

when passing option `--wait-ready` in CLI or `wait_ready` in process file, pm2 will wait for the application to `process.send('ready');` to set it as `started`

### CODE REVIEW

File: lib/God/ForkMode.js [hunk 1]
```diff
@@ -105,7 +105,6 @@ module.exports = function ForkMode(God) {
       cspr.process = {};
       cspr.process.pid = cspr.pid;
       cspr.pm2_env = pm2_env;
-      cspr.pm2_env.status = cst.ONLINE_STATUS;
```
Comment (Unitech) [inline lib/God/ForkMode.js]: why?

Comment (vmarchaud) [inline lib/God/ForkMode.js]: When the application is started, the status is assigned only after the application is online, not before.

Comment (soyuka) [inline lib/God/ForkMode.js]: Can't we keep this but put a `cst.LAUNCHING_STATUS` or something like this?

**Comment by Unitech** [comment]
it works well, we should add some sort of timeout to this feature. Like passing an optional timeout time to the `--wait-ready` option

**Pull request merged**